tools/firely-auth/firely-auth-settings.rst

```diff
@@ -0,0 +1,299 @@
+
+.. code-block:: json
+
+  "ClientRegistrationConfig": {
+      "AllowedClients": [
+          {
+              "ClientId": "Jv3nZkaxN36ucP33",
+              "ClientName": "Postman",
+              "Description": "Postman API testing tool",
+              "Enabled": true,
+              "RequireConsent": true,
+              "RedirectUris": ["https://www.getpostman.com/oauth2/callback", "https://oauth.pstmn.io/v1/callback"],
+              "ClientSecrets": [{"SecretType": "SharedSecret", "Secret": "re4&ih)+HQu~w"}], // SharedSecret, JWK
+              "AllowedGrantTypes": ["client_credentials", "authorization_code"],
+              "AllowedSmartLegacyActions": [],
+              "AllowedSmartActions": ["c", "r", "u", "d", "s"],
+              "AllowedSmartSubjects": [ "patient", "user", "system"],
+              "AlwaysIncludeUserClaimsInIdToken": true,
+              "RequirePkce": false,
+              "AllowOfflineAccess": false,
+              "AllowOnlineAccess": false,
+              "AllowFirelySpecialScopes": true,
+              "RequireClientSecret": true,
+              "LaunchIds": [],
+              "RefreshTokenLifetime": "30"
+          }
+      ]
+  }
+
+You register a :term:`client` in the ``AllowedClients`` array. For each client you can configure these settings:
+
+- ``ClientId``: string: unique identifier for this client. It should be known to the client as well
+- ``ClientName``: string: human readable name for the client, it is shown on the consent page
+- ``Description``: string:  human readable description of the client
+- ``Enabled``: true / false: simple switch to enable or disable a client (instead of removing it from the list)
+- ``RequireConsent``: true / false: when true, Firely Auth will show the user a page for consent to granting the requested scopes to the client, otherwise all requested and valid scopes will be granted automatically.
+- ``RedirectUris``: array of strings: url(s) on which Firely Auth will send the authorization code and access token. The client can specify one of the preregistered urls for a specific request.
+- ``ClientSecrets``: secrets can be of type ``SharedSecret`` or ``JWK``. You can have multiple of each, so you can accept two secrets for a short period of time to support key rotation and an update window for the client
+
+  - SharedSecret: ``{"SecretType": "SharedSecret", "Secret": "<a secret string shared with the client>"}`` - this can be used for either :term:`client credentials` or :term:`authorization code flow`, but only with a :term:`confidential client`.
+  - JWK: ``{"SecretType": "JWK", "SecretUrl": "<JWKS url>"}`` - where the JWKS url hosts a JSON Web Key Set that can be retrieved by Firely Auth, see also :term:`JWK`.
+  - JWK: ``{"SecretType": "JWK", "Secret": "<JWK>"}`` - where JWK is the contents of a :term:``JWK``. Use this if the client cannot host a url with a JWKS. 
```

Comment:
```suggestion
  - JWK: ``{"SecretType": "JWK", "Secret": "<JWK>"}`` - where JWK is the contents of a :term:`JWK`. Use this if the client cannot host a url with a JWKS. 
```

Otherwise,
<img width="627" alt="image" src="https://user-images.githubusercontent.com/873290/202433928-35ee229a-cdb0-49f8-8f67-96df714c6203.png">